src/main/scala/org/broadinstitute/hail/sparkextras/OrderedRDD.scala

```diff
@@ -38,72 +44,80 @@ object OrderedRDD {
   }
 
   def apply[PK, K, V](rdd: RDD[(K, V)], fastKeys: Option[RDD[K]] = None)(implicit kOk: OrderedKey[PK, K]): OrderedRDD[PK, K, V] = {
+    val (_, orderedRDD) = coerce(rdd, fastKeys)
+    orderedRDD
+  }
+
+  def coerce[PK, K, V](rdd: RDD[(K, V)], fastKeys: Option[RDD[K]] = None)
+    (implicit kOk: OrderedKey[PK, K]): (Int, OrderedRDD[PK, K, V]) = {
     import kOk.kct
     import kOk.pkct
     import kOk.kOrd
     import kOk.pkOrd
     import Ordering.Implicits._
 
     if (rdd.partitions.isEmpty)
-      return empty(rdd.sparkContext)
+      return (ORDERED_PARTITIONER, empty(rdd.sparkContext))
```

Comment:
I'm not sure.  It doesn't really matter, I guess.  I think I used AS_IS to indicate that you spent an action to check, but then you're done  The ORDERED_PARTITIONER really means that no work was done, which happens when the partitioner is ordered or the RDD is empty